### ISSUE TITLE
AVMediaType no longer available in net6.0-ios

### ISSUE THREAD

**Issue by Cheesebaron** [xamarin/xamarin-macios#16318]
### Steps to Reproduce

In Xamarin.iOS we could check for MediaType on a AVAssetTrack like so:

```csharp
if (track.MediaType == AVMediaType.Video)
   // do something
```

`AVMediaType` is no longer available in `net6.0-ios` and seems like it has been replaced with the enum `AVMediaTypes`. However, using that for comparison does not work as MediaType does not directly map into that enum.

This example doesn't work in `net6.0-ios` as `MediaType` would return "vide" and not "Video".

```csharp
if (track.MediaType == AVMediaTypes.Video.ToString())
  // do something
```

### Expected Behavior
AVMediaType static class being available for easy comparison with AVAssetTrack MediaType

### Actual Behavior
Not available and replaced with `AVMediaTypes` enum

### Environment

<details>
<summary>Version information</summary>

```
net6.0-ios

Installed Workload Ids      Manifest Version      Installation Source
---------------------------------------------------------------------
macos                       12.3.454/6.0.400      SDK 6.0.400
ios                         15.4.454/6.0.400      SDK 6.0.400
maccatalyst                 15.4.454/6.0.400      SDK 6.0.400
maui                        6.0.540/6.0.400       SDK 6.0.400
tvos                        15.4.454/6.0.400      SDK 6.0.400
android                     32.0.465/6.0.400      SDK 6.0.400
android-33                  32.0.465/6.0.400      SDK 6.0.400
```

</details>

**Comment by chamons** [comment]
This might be because https://github.com/xamarin/xamarin-macios/blob/main/src/avfoundation.cs#L3823-L3824 was not updated to return the smart enum.

I'm looking into where this change was made exactly. 

Using GetConstant on the smart enum to get the value for API usages that are bound as string is correct.

**Comment by chamons** [comment]
Yep, it was was done [here](https://github.com/xamarin/xamarin-macios/commit/ec045766cab664559725bdf42054e2073342080f).

And a number of tests were updated like this:

![image](https://user-images.githubusercontent.com/6392201/195379053-3144a080-bd29-4551-82cf-198145e549ce.png)

I'm going to leave this open to suggest adding smart enum support to that API. I'm not sure we can do that in an API safe way, thoughts @rolfbjarne ?

**Comment by rolfbjarne** [comment]
We can't do this in an API-compatible way (unless we expose using a different name, which we could certainly do).

This also seems to affect:

* AVAssetWriterInput.MediaType
* AVAssetTrack.MediaType
* AVMediaSelectionOption.MediaType
* AVCaptureInputPort.MediaType
* AVCaptureDeviceFormat.MediaType
* AVAssetSegmentTrackReport.MediaType
